event_reporter/reporter/application_event_reporter.go

```diff
@@ -425,6 +403,59 @@ func (s *applicationEventReporter) processResource(
 	return nil
 }
 
+func (s *applicationEventReporter) getResourceActualState(ctx context.Context, logCtx *log.Entry, metricsEventType metrics.MetricEventType, rs appv1.ResourceStatus, parentApplication *appv1.Application, childApplication *appv1.Application) (*application.ApplicationResourceResponse, error) {
+	if isApp(rs) {
+		if childApplication.TypeMeta.Size() == 0 || childApplication.TypeMeta.Kind == "" || childApplication.TypeMeta.APIVersion == "" {
+			// make sure there is type meta on object
+			childApplication.TypeMeta = metav1.TypeMeta{
+				Kind:       appv1reg.ApplicationKind,
+				APIVersion: appv1.SchemeGroupVersion.String(),
+			}
+		}
+
+		manifestBytes, _ := json.Marshal(childApplication)
```

Comment:
i think it is not make any sense to check if it is nil, i believe it is better to check error in such case

Reply:
we still ignore error